### ISSUE TITLE
Move websites to a dedicated branch

### ISSUE THREAD

**Issue by spoenemann** [eclipse/xtext#1076]
AFAIK both http://xtext.org and http://xtend-lang.org are currently generated from the maintenance branch, which still contains the whole old repository content. We should create a new branch and clean it so it contains only the websites.

**Comment by cdietrich** [comment]
@spoenemann @meysholdt since we now use master to build the aggregate repo should we use master for the web site as well or should that be a different branch?

**Comment by spoenemann** [comment]
Maybe we should have it both in master and in maintenance, and have the website be generated from maintenance as done currently. That would allow to write documentation for new features in master that will be visible after the release.

**Comment by svenefftinge** [comment]
+1 also this is how we used to do it and everything is setup for that.

**Comment by LorenzoBettini** [comment]
+1

**Comment by cdietrich** [comment]
@sponemann do you still have the scripts in place to allow the move of the 2 folders from maintenance to master?

**Comment by spoenemann** [comment]
Do you mean the repository splitting scripts? That's not applicable here. I would just copy the folders and reset the head of maintenance to master.

**Comment by cdietrich** [comment]
so no keeping history?

**Comment by spoenemann** [comment]
We should consider maintaining the websites in the new umbrella repo, see #1084.

**Comment by cdietrich** [comment]
+1 for umbrella

**Comment by svenefftinge** [comment]
Why do you want to spend time on changing the current approach?

**Comment by cdietrich** [comment]
- cause you have to clone bug fat xtext repo checkout maintenance to do edits in the documentation.
- we need to do this for each release to sync changes back to xtext-eclipse xtext-xtend help

**Comment by svenefftinge** [comment]
Ideally we would put the documentation into the individual projects they belong. I know that this is sometimes not easily possible as tutorials and explanations might cover multiple topics as they are use case centric. But the benefits would be documentation that sits more closely where the code is, and is also versioned together with the code. I think it would help us keeping code and docs in sync.

Still we would need a website repository (actually two) that composes that.

I'd rather spend time on this, than on moving the same big old documentation/website to another repository. User's typically don't clone it, as typos can be easily fixed online.

**Comment by cdietrich** [comment]
the cant, cause we are restricted by signing and eclipse cla

**Comment by spoenemann** [comment]
As a first step I cleaned up the maintenance branch like it was done on master after the splitting, but I left the websites folders (044ae63). I also merged the master cleanup point into maintenance with "ours" strategy (bec60c8). With this preparation it should be easy to merge maintenance into master after the 2.11 release and then reset maintenance to master. After that we'll have the website both in master and in maintenance, so we can write documentation of new features on master and do fixes on maintenance.

As a long term solution I would support the proposal of @svenefftinge to have local documentation for each project. However, we will first have to evaluate technical solutions to aggregate the separate documentations into the website. We should reconsider moving the website to xtext-umbrella in that context, but I'd leave it like it is for now.
